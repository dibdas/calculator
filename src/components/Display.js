/* eslint-disable react/default-props-match-prop-types */
/* eslint-disable react/require-default-props */
/* eslint-disable react/prop-types */
/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/no-typos */
import React from 'react';
import propTypes from 'prop-types';

const Display = (props) => (<div><h3>{props.result}</h3></div>);

Display.propTypes = {
  result: propTypes.string.isRequired,
};
Display.defaultProps = {
  result: '0',
};

export default Display;
