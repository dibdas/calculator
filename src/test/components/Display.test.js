import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Display from '../../components/Display';

describe('ButtonPanel', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Display />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the outcome given in as prop', () => {
    const { container } = render(<Display result="40" />);
    const card = container.querySelector('div');
    expect(card).toHaveTextContent('40');
  });
});
