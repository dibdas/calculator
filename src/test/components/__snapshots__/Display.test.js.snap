// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel renders correctly 1`] = `
<div>
  <h3>
    0
  </h3>
</div>
`;
